### PR TITLE
Ignore non-relevant search results (coordinates)

### LINKED ISSUE #1780: Search should not show non-relevant results after matching coordinates
image

### PR DESCRIPTION
**Pull request by trueTatar**
fixes #1780

### CODE REVIEW

**Comment by vng** [comment]
Thanks for the path!
To discuss: do we need additional results except coordinates say for "5 7", like house numbers, entries, or some other stuff ?

There are some fancy results, I do have now (this is debug desktop app):
Screenshot 2022-08-05 at 17 47 34

File: search/processor.cpp [hunk 1]
```diff
@@ -595,19 +595,21 @@ void Processor::Search(SearchParams const & params)
 
     try
     {
-      SearchDebug();
-      SearchCoordinates();
-      SearchPlusCode();
-      SearchPostcode();
-      if (viewportSearch)
+      if (bool coords_found = SearchCoordinates(); !coords_found)
```
Comment (biodranik) [inline search/processor.cpp:598]:
```suggestion
      if (!SearchCoordinates())
```

**Review by biodranik — APPROVED** [review]
A search query may contain building numbers like 5/1 or names like 7-11, these cases should not match coordinates.

If coordinates were entered and matched, other results are not needed.

File: search/processor.cpp (hunk 1 above)
Comment (vng) [reply search/processor.cpp:598]: "5 7" are coordinates. Additional search results from screenshot are relevant or not?
"-" is skipped in search processing as a delimiter

**Comment by vng** [comment]
"5 7" are coordinates. Additional search results from screenshot are relevant or not?
"-" is skipped in search processing as a delimiter

**Comment by biodranik** [comment]
> "-" is skipped in search processing as a delimiter

That was my concern. 5-7 should not be treated as a pair of coordinates. Can you please fix that?

**Comment by vng** [comment]
"5-7" like coordinates is a bug
Results from "5 7" looks relevant or not?

**Comment by biodranik** [comment]
1. Results on the screenshot are NOT relevant.
2. User expects to see these results when he/she types "5-7", "5/7", "5,7". These cases SHOULD work properly and do not convert into coordinates. Please create another issue for that if necessary.

**Comment by trueTatar** [comment]
- I have checked several values (typed by hand) and could not reproduce a bug, but if query contains new line symbol (e.g. '5 7\n') then number pair indeed is not treated as coordinates.
- I also have checked several separators (attaching screenshots with results) and all of them have been processed as expected except for hyphen. Looks like problem somewhere in function `MatchLanLonDegree`. I'm assuming that adding check for 'two numbers only' in the top of the function should resolve this hyphen issue.

**Review by vng — APPROVED** [review]
Ok, make sense. Please review comments.

**Review by vng — APPROVED** [review]

**Review by biodranik — APPROVED** [review]

**Pull request merged**